app/user/signup.tsx

```diff
@@ -0,0 +1,82 @@
+import { useState } from "react";
+
+export default function Signup({ onSignup }) {
+  const [username, setUsername] = useState("");
+  const [password, setPassword] = useState("");
+
+  return (
+    <>
+      <h1>Signup</h1>
+      <form autoComplete="off">
+        {" "}
+        <div>
+          {" "}
+          <label>Username: </label>{" "}
+          <input
+            id="username"
+            type="text"
+            value={username}
+            onChange={(e) => {
+              setUsername(e.currentTarget.value);
+            }}
+          ></input>{" "}
+        </div>{" "}
+        <br />{" "}
+        <div>
+          {" "}
+          <label>Password: </label>{" "}
+          <input
+            type="password"
+            id="password"
+            value={password}
+            onChange={(e) => {
+              setPassword(e.currentTarget.value);
+            }}
+          ></input>{" "}
+        </div>{" "}
+        <br />{" "}
+        <button type="button" onClick={validateInput}>
+          {" "}
+          Create{" "}
+        </button>{" "}
+      </form>
+    </>
+  );
+
+  function validateInput() {
+    var valid = true;
+    var errstyle = "2px solid red";
+    var initstyle = "1px solid grey";
+
+    if (username.length == 0) {
```

Comment:
```suggestion
    if (username.length === 0) {
```
This helps to avoid implicit type conversion, check [this](https://stackoverflow.com/questions/359494/which-equals-operator-vs-should-be-used-in-javascript-comparisons)